internal/server/server.go

```diff
@@ -178,16 +186,74 @@ func (s *Server) Stop() {
 	s.clientConnWg.Wait()
 
 	if err := s.poolMgr.Close(); err != nil {
-		s.logger.Printf("Error closing pool manager: %v\n", err)
+		s.logger.Error("Error closing pool manager", logr.Err(err))
 	}
 
 	if s.metricsSrv != nil {
 		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 		defer cancel()
 		if err := s.metricsSrv.Shutdown(ctx); err != nil {
-			s.logger.Printf("Error closing metrics server: %v\n", err)
+			s.logger.Error("Error closing metrics server", logr.Err(err))
 		}
 		// Wait until the ListenAndServer goroutine has finished.
 		s.metricsWg.Wait()
 	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	err := s.logger.Logr().ShutdownWithTimeout(ctx)
+	if err != nil {
+		// We have to resort to fmt because the logger is unavailable at this point.
+		fmt.Printf("Error while shutting down logger: %v\n", err)
+	}
+}
+
+// For simplicity's sake, please only use 3 logging levels:
+// info, warn and error. And let stack traces come with only panics.
```

Comment:
There's also `debug` in the `stringToStdLevel` function, should we list that in this comment too?

Reply:
nit, this is not a "please" condition, it's just forced :p